code/game/objects/items/devices/laserpointer.dm

```diff
@@ -193,16 +202,17 @@
 		to_chat(user, span_warning("Your fingers can't press the button!"))
 		return
 
-	if(!IN_GIVEN_RANGE(target, user, max_range))
-		to_chat(user, span_warning("\The [target] is too far away!"))
-		return
-	if(!(user in (view(max_range, target)))) //check if we are visible from the target's PoV
-		if(isnull(crystal_lens))
-			to_chat(user, span_warning("You can't point with [src] through walls!"))
-			return
-		if(!((user.sight & SEE_OBJS) || (user.sight & SEE_MOBS))) //only let it work if we have xray or thermals. mesons don't count because they are easier to get.
-			to_chat(user, span_notice("You can't quite make out your target and you fail to shine at it."))
+	if(max_range != -1)
+		if(!IN_GIVEN_RANGE(target, user, max_range))
+			to_chat(user, span_warning("\The [target] is too far away!"))
 			return
+		if(!(user in (view(max_range, target)))) //check if we are visible from the target's PoV
```

Comment:
Is this doing `view(INFINITY)`? 🥴

Reply:
No because it's only ran if max range isn't infinity

Reply:
![image](https://github.com/tgstation/tgstation/assets/51863163/97a0157e-20c4-4f3d-9d78-2bed7ab9b9bb)
?

Reply:
I can swap it to the define

![image](https://github.com/tgstation/tgstation/assets/53777086/e0b6cb6f-6eb0-48b6-aa8e-69a582a351cd)

Reply:
It confuses me when people use INFINITE and not INFINITY